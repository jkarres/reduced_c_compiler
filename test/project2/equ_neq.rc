
function : void main() {
  int i = 6, j = 7;
  float f = 6.0, g = 7.0;

  // int vs int

  cout << 1 << " " << i == i << endl;
  cout << 2 << " " << i == j << endl;
  cout << 3 << " " << j == i << endl;
  cout << 4 << " " << j == j << endl;

  cout << 5 << " " << i != i << endl;
  cout << 6 << " " << i != j << endl;
  cout << 7 << " " << j != i << endl;
  cout << 8 << " " << j != j << endl;

  // float vs int

  cout << 9 << " " << f == f << endl;
  cout << 10 << " " << f == j << endl;
  cout << 11 << " " << j == f << endl;
  cout << 12 << " " << j == j << endl;

  cout << 13 << " " << f != f << endl;
  cout << 14 << " " << f != j << endl;
  cout << 15 << " " << j != f << endl;
  cout << 16 << " " << j != j << endl;

  // int vs float

  cout << 17 << " " << i == i << endl;
  cout << 18 << " " << i == g << endl;
  cout << 19 << " " << g == i << endl;
  cout << 20 << " " << g == g << endl;

  cout << 21 << " " << i != i << endl;
  cout << 22 << " " << i != g << endl;
  cout << 23 << " " << g != i << endl;
  cout << 24 << " " << g != g << endl;

  // float vs float

  cout << 25 << " " << f == f << endl;
  cout << 26 << " " << f == g << endl;
  cout << 27 << " " << g == f << endl;
  cout << 28 << " " << g == g << endl;

  cout << 29 << " " << f != f << endl;
  cout << 30 << " " << f != g << endl;
  cout << 31 << " " << g != f << endl;
  cout << 32 << " " << g != g << endl;
}
