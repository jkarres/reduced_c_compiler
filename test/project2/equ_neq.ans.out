1 true
2 false
3 false
4 true
5 false
6 true
7 true
8 false
9 true
10 false
11 false
12 true
13 false
14 true
15 true
16 false
17 true
18 false
19 false
20 true
21 false
22 true
23 true
24 false
25 true
26 false
27 false
28 true
29 false
30 true
31 true
32 false
